libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -1334,25 +1335,101 @@ string YulUtilFunctions::allocationFunction()
 	});
 }
 
-string YulUtilFunctions::allocateMemoryArrayFunction(ArrayType const& _type)
+string YulUtilFunctions::allocateMemoryArrayFunction(ArrayType const& _type, bool _shouldZero)
 {
 	solUnimplementedAssert(!_type.isByteArray(), "");
 
-	string functionName = "allocate_memory_array_" + _type.identifier();
+	string functionName = "allocate_" + string(_shouldZero ? "and_zero_" : "") + "memory_array_" + _type.identifier();
+
+	if (_type.baseType()->hasSimpleZeroValueInMemory())
```

Comment:
I think it would be more uniform to stick to directly calling `m_functionCollector.createFunction` and then doing the case distinction inside - unless you never call it and resort to directly returning other `YulUtilFuncitons::...Function(`-calls.

Reply:
Also, the two cases are rather similar. Maybe they could keep a common "allocate+store length" and just call different inner functions inside?

Reply:
Done